spring-integration-r2dbc/src/test/java/org/springframework/integration/r2dbc/inbound/R2dbcMessageSourceTests.java

```diff
@@ -0,0 +1,169 @@
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+/**
+ * @author Rohan Mukesh
+ *
+ * @since 5.4
+ */
+@SpringJUnitConfig
+@DirtiesContext
+public class R2dbcMessageSourceTests {
+
+	@Autowired
+	DatabaseClient client;
+
+	R2dbcEntityTemplate entityTemplate;
+
+	@Autowired
+	R2dbcMessageSource r2dbcMessageSourceSelectOne;
+
+	@Autowired
+	R2dbcMessageSource r2dbcMessageSourceSelectMany;
+
+	@Autowired
+	R2dbcMessageSource r2dbcMessageSourceError;
+
+	@BeforeEach
+	public void setup() {
+		Hooks.onOperatorDebug();
+		entityTemplate = new R2dbcEntityTemplate(this.client);
+		List<String> statements = Arrays.asList(
+				"DROP TABLE IF EXISTS person;",
+				"CREATE table person (id INT AUTO_INCREMENT NOT NULL, name VARCHAR2, age INT NOT NULL);");
+
+		statements.forEach(it -> this.client.execute(it)
+				.fetch()
+				.rowsUpdated()
+				.as(StepVerifier::create)
+				.expectNextCount(1)
+				.verifyComplete());
+	}
+
+	@Test
+	public void validateSuccessfulQueryWithoutSettingExpectedElement() {
+		this.entityTemplate.insert(new Person("Bob", 35))
+				.then()
+				.as(StepVerifier::create)
+				.verifyComplete();
+
+		StepVerifier.create((Flux<?>) r2dbcMessageSourceSelectMany.receive().getPayload())
+				.assertNext(person -> assertThat(((Person) person).getName()).isEqualTo("Bob"))
+				.verifyComplete();
+	}
+
+
+	@Test
+	public void validateSuccessfulQueryWithSingleElementOfMonoDBObject() {
+		this.entityTemplate.insert(new Person("Bob", 35))
+				.then()
+				.as(StepVerifier::create)
+				.verifyComplete();
+
+		r2dbcMessageSourceSelectOne.setExpectSingleResult(true);
+
+		StepVerifier.create((Mono<?>) r2dbcMessageSourceSelectOne.receive().getPayload())
+				.assertNext(person -> assertThat(((Person) person).getName()).isEqualTo("Bob"))
+				.verifyComplete();
+
+	}
+
+	@Test
+	public void validateSuccessfulQueryWithMultipleElementOfFluxDBObject() {
+		this.entityTemplate.insert(new Person("Bob", 35))
+				.then()
+				.as(StepVerifier::create)
+				.verifyComplete();
+
+		this.entityTemplate.insert(new Person("Tom", 40))
+				.then()
+				.as(StepVerifier::create)
+				.verifyComplete();
+
+		StepVerifier.create((Flux<?>) r2dbcMessageSourceSelectMany.receive().getPayload())
+				.assertNext(person -> assertThat(((Person) person).getName()).isEqualTo("Bob"))
+				.assertNext(person -> assertThat(((Person) person).getName()).isEqualTo("Tom"))
+				.verifyComplete();
+
+	}
+
+	@Test
+	public void testAnyOtherObjectQueryExpression() {
+
+		StepVerifier.create((Flux<?>) r2dbcMessageSourceError.receive().getPayload())
+				.expectErrorMatches(throwable -> throwable instanceof IllegalStateException
+						&& throwable.getMessage().contains("'queryExpression' must evaluate to String or"))
+				.verify();
+	}
+
+	@Configuration
+	@Import(R2dbcDatabaseConfiguration.class)
+	static class R2dbcMessageSourceConfiguration {
+
+		@Autowired
+		DatabaseClient databaseClient;
+
+		@Bean(name = "r2dbcMessageSourceSelectOne")
```

Comment:
FYI: the method name becomes a bean name by default.
So, probably for this use-case you definitely don't need a `name` attribute on this annotation...